src/utils/sentry.ts

```diff
@@ -61,17 +61,49 @@ function initSentry() {
   // maskeAllText: mask all text on the screen.
   // ignoreErrors: Sentry ignores errors containing strings or regex. Regex needs to be exact message, string is partial
   // replaysSessionSampleRate: replays that begin recording immediately and last the entirety of the user's session.
-
+  // transport is what allows consumed apps to tell chrome what dsn to send things too.
   Sentry.init({
-    dsn: API_KEY,
+    //default api_key -> cp-001-insights
+    dsn: 'https://8b6372cad9604745ae3606bc4adc0060@o271843.ingest.sentry.io/1484024',
     environment: `Prod${appDetails.beta}`,
     maxBreadcrumbs: 50,
     attachStacktrace: true,
-    integrations: [new BrowserTracing(), new Sentry.Replay({ maskAllText: false, maskAllInputs: true })],
+    integrations: [
+      Sentry.browserTracingIntegration(),
+      Sentry.replayIntegration({ maskAllText: false, maskAllInputs: true }),
+      SentryBrowser.moduleMetadataIntegration(),
+    ],
     tracesSampleRate: 0.2,
     debug: !!window.localStorage.getItem('chrome:sentry:debug'),
     replaysOnErrorSampleRate: 1.0,
     replaysSessionSampleRate: 0.3,
+    transport,
+    beforeSend: (event) => {
+      if (event?.exception?.values?.[0]?.stacktrace?.frames) {
+        const frames = event.exception.values[0].stacktrace.frames;
+        // Find the last frame with module metadata containing a DSN
+        const routeTo = frames
+          .filter((frame) => frame.module_metadata && frame.module_metadata.dsn)
+          .map((v) => v.module_metadata)
+          .slice(-1); // using top frame only - you may want to customize this according to your needs
+
+        if (routeTo.length) {
+          event.extra = {
+            ...event.extra,
+            [EXTRA_KEY]: routeTo,
+          };
+        }
+      }
+
+      return event;
+    },
+    //These two apps will not be set up as of now. This helps limit transacations
+    beforeSendTransaction: (event) => {
+      if (event?.contexts?.app?.app_name === 'subscriptions' || event.contexts?.app?.app_name === 'image-builder') {
```

Comment:
I'd like these to be in some variables, potentially configured via Chrome service. Store it in an array for now, and check if the app_name value is included.